### ISSUE TITLE
Include check rules of SCA policies into policy database

### ISSUE THREAD

**Issue by chemamartinez** [wazuh/wazuh#2960]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.9.0 | Manager/Agent | Packages/Sources | All |

Here we can see a check of an SCA policy:

```
- id: 5031
   title: "Ensure IP forwarding is disabled"
   description: "The net.ipv4.ip_forward and net.ipv6.conf.all.forwarding flags are used to tell the system whether it can forward packets or not."
   rationale: "Setting the flags to 0 ensures that a system with multiple interfaces (for example, a hard proxy), will never be able to forward packets, and therefore, never serve as a router."
   remediation: "Set the following parameter in /etc/sysctl.conf or a /etc/sysctl.d/* file: net.ipv4.ip_forward = 0, net.ipv6.conf.all.forwarding = 0"
   compliance:
    - cis_csc: "5.1"
    - cis: "3.1.1"
   condition: any
   rules:
     - 'f:/proc/sys/net/ipv4/ip_forward -> 1;'
     - 'f:/proc/sys/net/ipv6/ip_forward -> 1;'
```

The section `rules` is missing in events, alerts and the database where all the data is stored for checks and policies. Instead, just the affected files/processes/registries/directories by the rule were being stored.

We have to replace those fields (files, processes, etc.) by a new table which store all the rules executed for each check of a policy. It will be defined as follows:

```
CREATE TABLE IF NOT EXISTS sca_check_rules (
   id_check INTEGER REFERENCES sca_check (id),
  type TEXT,
  rule TEXT,
  PRIMARY KEY (id_check, type, rule)
);
```

For example, for the previous check, the table entries would be:

```
5031 | file | f:/proc/sys/net/ipv4/ip_forward -> 1;
5031 | file | f:/proc/sys/net/ipv6/ip_forward -> 1;
```

### Involved tasks

- [x] Add the rules in every check event.

- [x] Decode the `rules` fields on the manager side and store them in the DB.

- [x] Add to the DB schema and queries the new table and fields.

- [x] Purge the new table when the check integrity of the DB fails.

- [ ] Documentation to make easier to understand the SCA rules syntax.

**Comment by Skeptor** [comment]
Currently I'm working on branch: https://github.com/wazuh/wazuh/tree/add_sca_rules_bd

This branch has a commit: https://github.com/wazuh/wazuh/commit/5d8ade34322d9796ce36f4d20f0cacc333885070 to remove the type of rule from DB.
The next step is to send the rules in the SCA event as an array of rules to store it in the DB.

**Comment by chemamartinez** [comment]
Merged: #3076
